File: StriverA2Z/Java/BinaryTrees/PathSum.java
public class PathSum {
    public static void main(String[] args) {
        Node root = BinaryTreeOperations.arrayToBTLevelOrder(new Integer[]{10, 20, 30, 40, 50, 60, 70});
        int pathSum = pathSum(root);
        System.out.println(pathSum);
    }

    public static int pathSum(Node root) {
        int[] sum = new int[1];
        sum[0] = Integer.MIN_VALUE;
        heightOfBT(root, sum);
        return sum[0];
    }

    public static int heightOfBT(Node root, int[] sum) {
        if (root == null) return 0;
        int left = Math.max(0, heightOfBT(root.left, sum));
        int right = Math.max(0, heightOfBT(root.right, sum));
        sum[0] = Math.max(sum[0], left + right + root.data);
        return root.data + Math.max(left, right);
    }
}

File: StriverA2Z/JavaScript&TypeScript/src/BinaryTrees/pathsum.ts
import {arrayToBTLevelOrder, BTNode} from "./BinaryTreeOperations";

function pathsum(root: BTNode) {
	const sum = [Number.MIN_SAFE_INTEGER];
	height(root, sum);
	return sum[0];
}

function height(root: BTNode, sum: number[]) {
	if (root === null) return 0;
	const left = Math.max(0, height(root.left, sum));
	const right = Math.max(0, height(root.right, sum));
	sum[0] = Math.max(sum[0], left + right + root.data);
	return root.data + Math.max(left, right);
}

let root = arrayToBTLevelOrder([10,20,30,40,50,60,70]);
console.log(pathsum(root));
